DatadogCore/Sources/Core/DatadogCore.swift

```diff
@@ -159,8 +159,15 @@ internal final class DatadogCore {
     ///
     /// - Parameter trackingConsent: new consent value, which will be applied for all data collected from now on
     func set(trackingConsent: TrackingConsent) {
+        // There could be events in the queue that were collected before the consent change
+        // but not written to the disk yet, which must be migrated to the new consent.
+        // Flush ensures we are in a consistent state before the consent change.
+        flush()
```

Comment:
Do we want to flush the data whatever the consent value is?

Reply:
N/A anymore with the new fix we do it without flushing